test/Microsoft.AspNetCore.SignalR.Tests/TestClient.cs

```diff
@@ -25,7 +25,7 @@ public class TestClient : IDisposable, IInvocationBinder
 
         public DefaultConnectionContext Connection { get; }
         public Channel<byte[]> Application { get; }
-        public Task Connected => Connection.Metadata.Get<TaskCompletionSource<bool>>("ConnectedTask").Task;
+        public Task Connected => (Connection.Metadata["ConnectedTask"] as TaskCompletionSource<bool>).Task;
```

Comment:
why `as` casting? Implies that ConnectedTask might not be a `TaskCompletionSource<bool>`. If this was true we should handle this.